### PR TITLE
Fix mooseWarnings in phase field and tensor mechanics

### PR DESCRIPTION
**Pull request by dschwen**
This enables us to run the tests in phase field and tensor mechanics with `--error`.

Refs #6692

### CODE REVIEW

File: modules/tensor_mechanics/tests/linear_elastic/tests [hunk 1]
```diff
@@ -1,10 +1,4 @@
 [Tests]
-  [./tween]
-    type = 'Exodiff'
-    input = 'tween.i'
```
Comment (permcody) [inline modules/tensor_mechanics/tests/linear_elastic/tests]: Can we remove this input file and the gold file too?

Comment (dschwen) [inline modules/tensor_mechanics/tests/linear_elastic/tests]: I'd rather not. I'm referring to this file in the deprecation documentation

http://mooseframework.org/wiki/PhysicsModules/TensorMechanics/Deprecations/LinearElasticMaterial/

Comment (permcody) [inline modules/tensor_mechanics/tests/linear_elastic/tests]: OK - but eventually you will remove it. I can live with that. I'm just concerned that it'll get out of date quickly.

Comment (dschwen) [inline modules/tensor_mechanics/tests/linear_elastic/tests]: I will remove it when `LinearElasticMaterial` is removed. I want to make the migration path as clear and simple as possible.

**Pull request merged**